src/middlewared/middlewared/service.py

```diff
@@ -871,6 +873,67 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, timeout, counter=3):
+        if osc.IS_LINUX:
+            process = run(['ping', '-w', f'{timeout}', '-c', f'{counter}', host])
+        else:
+            process = run(['ping', '-t', f'{timeout}', '-c', f'{counter}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    def _ping6_host(self, host, timeout, counter=3):
+        if osc.IS_LINUX:
+            process = run(['ping6', '-w', f'{timeout}', '-c', f'{counter}', host])
+        else:
+            process = run(['ping6', '-X', f'{timeout}', '-c', f'{counter}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Dict(
+            'options',
+            Str('type', enum=['ICMP', 'ICMPV4', 'ICMPV6'], default='ICMP'),
+            Str('hostname', required=True),
+            Int('timeout', validators=[Range(min=1, max=60)], default=10),
+        ),
+    )
+    def ping_remote(self, options):
+        """
+        Method that will send an ICMP echo request to "hostname"
+        and will wait up to "timeout" for a reply.
+        """
+        ip = None
+        ip_found = True
+        try:
+            ip = IpAddress()
+            ip(options['hostname'])
+            ip = options['hostname']
+        except ValueError:
+            ip_found = False
+        if not ip_found:
+            try:
+                ip = socket.getaddrinfo(options['hostname'], None)[0][4][0]
```

Comment:
I dont think you're doing the right thing still.

If you specify ICMPV6 it should force resolving to IPV6, right now it doesnt. It will just fail if it resolves to a IPV4 by default.